Primo.js

```diff
@@ -197,18 +197,20 @@ function fetchPNX(itemData) {
 	Z.debug("Trying " + url);
 	ZU.doGet(url,
 		function (text) {
-			text = text.trim();
-			if (text.substr(0, 5) != '<?xml' || text.search(/<error\b/i) !== -1) {
-				// try a different PNX url
-				gotPNX = false;
-				return;
-			}
-			else {
-				gotPNX = true;
-				PNXUrlGenerator.confirmed = true;
+			if (text) {
```

Comment:
Can you explain a situation where this occurs?

Reply:
I found that problem on test in Primo.js:
http://princeton-primo.hosted.exlibrisgroup.com/primo_library/libweb/action/dlDisplay.do?vid=PRINCETON&search_scope=All%20Princeton%20Libraries&docId=PRN_VOYAGER2778598&fn=permalink
The error message was: TypeError: Cannot read property 'trim' of undefined
This solution is for the next row, when it try trim for undefined 'text' paramter - text = text.trim();
(Primo.js throw this error but still continue to try import the PNX record.)

Reply:
I'm asking when `text` would be undefined — I'm not sure why that would happen, unless maybe the connection itself was interrupted. I'm not seeing that on the linked page.

Reply:
Not sure about that, i removed it and the tests run successfully.